src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -46,6 +47,11 @@ case class CsvRelation protected[spark] (
     inferCsvSchema: Boolean)(@transient val sqlContext: SQLContext)
   extends BaseRelation with TableScan with InsertableRelation {
 
+  /**
+   * Limit the number of lines we'll search for a header row that isn't comment-prefixed.
+   */
+  private val MAX_COMMENT_LINES_IN_HEADER = 100
```

Comment:
I feel 100 lines may be excessive. How many comment lines do you usually see in your use cases? Alternatively, we can set a smaller default and expose it as an option to be increased.